sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/functions/FunctionsTest.java

```diff
@@ -1092,4 +1092,55 @@ public void testTruncateBigInt() {
                 5678.1"""
         );
     }
+
+    @Test
+    public void testExp() {
+        this.qs("""
+                SELECT EXP(0.0);
+                 exp
+                -----
+                 1.0
+                (1 row)
+                
+                SELECT EXP(0e0);
+                 exp
+                -----
+                 1.0
+                (1 row)
+                
+                SELECT EXP(null);
+                 exp
+                -----
+                null
+                (1 row)
+                
+                SELECT EXP(null::double);
+                 exp
+                -----
+                null
+                (1 row)
+                
+                SELECT EXP(0);
+                 exp
+                -----
+                 1
+                (1 row)
+                """
+        );
+    }
+
+    @Test @Ignore("FP comparison error for Calcite optimized version")
+    public void testExpEdgeCase() {
+        this.qs("""
+                -- changed the type from numeric to double
+                -- therefore the value has also slightly changed
+                -- cases that used to generate inaccurate results in Postgres
+                select exp(32.999::double);
+                         exp
+                ---------------------
+                 214429043492155.56
+                (1 row)
+                """
+        );
+    }
```

Comment:
```rust
error: test failed, to rerun pass `--lib`
stdout ----
R: ("214429043492155.5625",)x1
L: ("214429043492155.59375",)x1
thread 'test0' panicked at src/lib.rs:151:5:

    mvn test -Dtest=FunctionsTest#testExpEdgeCase
    -- changed the type from numeric to double
    -- therefore the value has also slightly changed
    -- cases that used to generate inaccurate results in Postgres
    select exp(32.999::double)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    test0

test result: FAILED. 1 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

Both values are pretty close. 

And when compared with [float-cmp](https://docs.rs/float-cmp/latest/float_cmp/) are approximately equal, between 1 ULP. 

```rust
    let expected = 214429043492155.5625_f64;
    let got = 214429043492155.59375_f64;
    assert!(expected.approx_eq(got, (f64::EPSILON, 1)));

```

Reply:
Can you check if they are from calcite? Then it's the bug I fixed there

Reply:
This is for the Calcite optimized version. `214429043492155.59375` is what Calcite returns. 

```rust
// generated optimized code
Tup1::new(F64::new(f64::from_bits(4821209731050809203u64))/*2.144290434921556E14*/)

// the output it is being compared against
Tup1::new(F64::new(f64::from_bits(4821209731050809202u64))/*2.1442904349215556E14*/)
```